### PR TITLE
Cross-check username and password similarity

### PR DESCRIPTION
**Pull request by nicolearagao**
**Summary**
This PR adds dynamic similarity validation between the admin username and password during their respective reset operations. Instead of checking against a hardcoded "admin" value, the system now fetches the actual stored secret and validates that the new value is not too similar to its counterpart.

**Changes**
_podman_utils: Add get_secret_value function (ea76ceb)_
1. New function to retrieve the actual value of an existing podman secret
2. Added redact_output parameter to run_command() to prevent sensitive values from appearing in logs

_secrets: Add build_similar_value_check helper (70ac752)_
1. Helper function that fetches a podman secret and wraps it in a SimilarValueCheck object
2. Returns None if the secret doesn't exist, allowing callers to skip the similarity check gracefully

_reset_admin_password: Add dynamic similarity check (4fc6ab8)_
1. Replaced hardcoded "admin" similarity check with dynamic username lookup
2. Now fetches the actual stored username secret for validation
3. Skips the similarity check if no username has been set yet

_reset_admin_username: Add similarity check (ebce464)_

1. Added password similarity validation when setting a new username
2. Ensures the username is not too similar to the current password
3. Skips the check if no password has been set yet

## Summary by Sourcery

Enforce dynamic similarity checks between the admin username and password when resetting credentials, and add secure podman secret retrieval utilities.

New Features:
- Add dynamic similarity validation so new admin passwords are checked against the current username and new admin usernames are checked against the current password.
- Introduce a helper to build similarity checks from existing podman secrets, skipping validation gracefully when the counterpart secret is missing.
- Add a podman utility to retrieve existing secret values while supporting redacted command output logging.

Enhancements:
- Allow username reset helpers to accept configurable check requirements and reuse the common secret validation logic.
- Extend the generic command runner to optionally redact stdout and stderr from logs for sensitive operations.

Tests:
- Add unit tests covering similarity-check behavior when resetting admin usernames and passwords, including accept and reject paths.
- Add tests for the new podman secret retrieval helper, including error handling, argument validation, and whitespace preservation.
- Add tests for building similarity checks from secrets, including the case where the underlying secret is absent.

### CODE REVIEW

File: src/quipucordsctl/podman_utils.py [hunk 1]
```diff
@@ -308,3 +308,33 @@ def pull_image(image_id: str, wait_timeout: int = None) -> bool:
...
+    """
+    verify_podman_argument_string(_("podman secret name"), secret_name)
+    if not secret_exists(secret_name):
+        return None
+    stdout, __, exit_code = shell_utils.run_command(
+        [
+            "podman",
+            "secret",
+            "inspect",
+            "--showsecret",
```
Comment (nicolearagao) [inline src/quipucordsctl/podman_utils.py]: Evaluated other options, this is the most straightforward. THis should be compatible with RHEL 8.x and later

File: src/quipucordsctl/podman_utils.py [hunk 2]
```diff
@@ -308,3 +308,33 @@ def pull_image(image_id: str, wait_timeout: int = None) -> bool:
...
+            "--format",
+            "{{.SecretData}}",
+            secret_name,
+        ],
```
Comment (nicolearagao) [inline src/quipucordsctl/podman_utils.py]: should we have any security concerns ?

File: src/quipucordsctl/secrets.py [hunk 3]
```diff
@@ -347,6 +358,27 @@ class SimilarValueCheck:
     max_similarity: float = 1.0
 
 
+def build_similar_value_check(
+    secret_name: str,
+    display_name: str,
+    max_similarity: float = 0.7,
```
Comment (infinitewarp) [inline src/quipucordsctl/secrets.py]: Just sharing a little historic context on this `0.7` magic number. When I wrote the first iteration of the value check using the `MAX_SIMILARITY` constant, I was drawing inspiration from this: https://github.com/django/django/blob/0a983330cd57dd2bb1dfa91a5a3ed742559eebe6/django/contrib/auth/password_validation.py#L177

**Review by infinitewarp — CHANGES_REQUESTED** [review]
I am just planting "request changes" here so we don't forget to fix the logging issue you spotted in `shell_utils.run_command` that we also discussed in Slack.

File: tests/commands/test_reset_admin_username.py [hunk 4]
```diff
@@ -219,3 +219,111 @@ def test_reset_admin_username_prompt_returns_none(mocker, caplog):
...
+    assert reqs["min_length"] == 1
+    assert reqs["digits"] is False
+    assert reqs["letters"] is False
+    assert reqs["not_isdigit"] is False
+
+
+def test_reset_admin_username_rejects_similar_to_password(
+    mock_first_time_run, mocker, caplog
+):
+    """Test username is rejected when too similar to existing password."""
```
Comment (mirekdlugosz) [inline tests/commands/test_reset_admin_username.py]: It feels strange to have basically the same tests in two different places, not sharing anything.

I guess it makes sense given that both username and password may come from environment variable or pre-existing secret, so you can never be sure which one will be provided first. But I would hope the code makes it more deterministic at some point.

**Review by mirekdlugosz — APPROVED** [review]
Left some suggestions / thoughts. Thinking about it, it's probably not blocking and we can take some later in a followup work.

File: src/quipucordsctl/shell_utils.py
Comment (infinitewarp) [inline src/quipucordsctl/shell_utils.py]: As of the current commit at the time I'm posting this comment, these assignments to `stdout` and `stderr` are still shadowing/replacing the incoming `stdout` and `stderr` function arguments, and we should rename these (or those) to ensure readability of the code and to avoid unintended side effects.

**Review by infinitewarp — APPROVED** [review]
I am approving this PR, but please note the comment I left indicating that we are still shadowing `stdout` and `stderr` which I thought we discussed offline we would change/fix here. This does not need to be a blocking request for change, though.

**Review by infinitewarp — APPROVED** [review]
Approving again to disregard my earlier comment about variable shadowing. Thank you for fixing that! 😉
